File: intrinsic/util/path_resolver/path_resolver_test.py
# Copyright 2023 Intrinsic Innovation LLC

"""Tests for helper functions to resolve runfiles paths."""

import os

from absl.testing import absltest
from intrinsic.util.path_resolver import path_resolver

TEST_FILE = 'intrinsic/util/path_resolver/path_resolver_test.py'


class PathResolverTest(absltest.TestCase):

  def test_resolve_runfiles_path(self):
    path = path_resolver.resolve_runfiles_path(TEST_FILE)
    self.assertTrue(os.path.exists(path))


if __name__ == '__main__':
  absltest.main()

File: intrinsic/util/path_resolver/path_resolver_test.cc
// Copyright 2023 Intrinsic Innovation LLC

#include "intrinsic/util/path_resolver/path_resolver.h"

#include <gtest/gtest.h>

#include <string>
#include <string_view>

namespace intrinsic {
namespace {

TEST(PathResolver, Resolve) {
  std::string run_file = "intrinsic/util/path_resolver/path_resolver_test.cc";

  std::string full_path =
      PathResolver::ResolveRunfilesPath(std::string_view(run_file));
  EXPECT_NE(full_path.size(), 0);
  EXPECT_NE(full_path, run_file);
}

TEST(PathResolver, ResolveRunfilesPathIfRelative) {
  EXPECT_EQ(PathResolver::ResolveRunfilesPathIfRelative("/absolute/path"),
            "/absolute/path");
  EXPECT_EQ(PathResolver::ResolveRunfilesPathIfRelative("relative/path"),
            PathResolver::ResolveRunfilesPath("relative/path"));
}

}  // namespace
}  // namespace intrinsic
